### PR TITLE
Better head vs camera checks for avatar head cauterization

### PR DESCRIPTION
**Pull request by hyperlogic**
* cameraInsideHead() check now uses detailed avatar collision when possible.
* head is now more constantly hidden in first person camera mode
* getEyeModelPositions() uses a better estimate when avatar eye joints are missing.
* moved findPointKDopDisplacement from Rig.cpp into AnimUtil.cpp
* added isPlayingOverrideAnimation() method to Rig class

https://highfidelity.manuscript.com/f/cases/20494/Camera-is-places-in-wired-position-when-avatar-has-no-eye-joints
https://highfidelity.manuscript.com/f/cases/21400/Sometimes-the-cauterization-doesn-t-work-correctly

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18671/HighFidelity-Beta-PR15365-02d5769.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18671/HighFidelity-Beta-PR15365-02d5769.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18671/HighFidelity-Quest-Beta-PR15365-02d5769.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18671/HighFidelity-Beta-PR15365-02d5769.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18671/HighFidelity-Beta-Interface-PR15365-02d5769.exe

File: libraries/avatars-renderer/src/avatars-renderer/SkeletonModel.cpp [hunk 1]
```diff
@@ -270,28 +270,13 @@ bool SkeletonModel::getEyeModelPositions(glm::vec3& firstEyePosition, glm::vec3&
...
-        const float EYE_SEPARATION = 0.1f;
-        firstEyePosition = baseEyePosition + headRotation * glm::vec3(EYE_SEPARATION, 0.0f, EYES_FORWARD_HEAD_ONLY);
-        secondEyePosition = baseEyePosition + headRotation * glm::vec3(-EYE_SEPARATION, 0.0f, EYES_FORWARD_HEAD_ONLY);
+
+    glm::vec3 headPosition;
+    if (getJointPosition(_rig.indexOfJoint("Head"), headPosition)) {
+        float heightRatio = _rig.getUnscaledEyeHeight() / DEFAULT_AVATAR_EYE_HEIGHT;
+        glm::vec3 ipdOffset = glm::vec3(DEFAULT_AVATAR_IPD / 2.0f, 0.0f, 0.0f);
+        firstEyePosition = headPosition + heightRatio * (DEFAULT_AVATAR_HEAD_TO_MIDDLE_EYE_OFFSET + ipdOffset);
+        secondEyePosition = headPosition + heightRatio * (DEFAULT_AVATAR_HEAD_TO_MIDDLE_EYE_OFFSET - ipdOffset);
```
Comment (amantley) [inline libraries/avatars-renderer/src/avatars-renderer/SkeletonModel.cpp]: This is much better than the old version, where the scale was base on 1 meter.  My only question is why we don't need to multiply the head to eye offset by the head rotation.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18690/HighFidelity-Beta-PR15365-7aeb4bc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18690/HighFidelity-Beta-Interface-PR15365-7aeb4bc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18690/HighFidelity-Beta-PR15365-7aeb4bc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18690/HighFidelity-Beta-Interface-PR15365-7aeb4bc.exe

**Comment by hyperlogic** [comment]
android build failed, retrying.
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18695/HighFidelity-Beta-PR15365-7aeb4bc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18695/HighFidelity-Beta-Interface-PR15365-7aeb4bc.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18695/HighFidelity-Beta-PR15365-7aeb4bc.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18695/HighFidelity-Quest-Beta-PR15365-7aeb4bc.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18695/HighFidelity-Beta-PR15365-7aeb4bc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18695/HighFidelity-Beta-Interface-PR15365-7aeb4bc.exe

**Pull request merged**